javascript/tables/src/validationUtils.ts

```diff
@@ -0,0 +1,324 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { ValidationTestResult } from '@datashaper/schema'
+import { DataType, ErrorCode } from '@datashaper/schema'
+
+export function validateRequiredConstraint(): (
+	values: unknown,
+	includeInstances: boolean,
+) => ValidationTestResult {
+	return function validateRequiredConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const valuesCasted = values as string[]
+
+		if (!includeInstances) {
+			return {
+				fail: !valuesCasted.every(validateRequired),
+				indexes: [],
+				rule: ErrorCode.Required,
+			}
+		} else {
+			const resultIndexes: number[] = []
+
+			valuesCasted.map((value: string, index: number) => {
+				if (!validateRequired(value)) resultIndexes.push(index)
+			})
+
+			return {
+				fail: resultIndexes.length != 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Required,
+			}
+		}
+	}
+}
+
+export function validateRequired(element: string): boolean {
+	if (element == null || element == undefined || element.length == 0)
+		return false
+
+	return true
+}
+
+export function validateUniqueConstraint(): (
+	values: unknown,
+	includeInstances: boolean,
+) => ValidationTestResult {
+	return function validateUniqueConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const uniqueValues = new Set<string>()
+		const resultIndexes: number[] = []
+		const valuesCasted = values as string[]
+
+		valuesCasted.map((value: string, index: number) => {
+			if (uniqueValues.has(value)) {
+				if (includeInstances) resultIndexes.push(index)
+			} else {
```

Comment:
This can be simplified a bit: since it is a Set, you can just add the value every time, so you don't need this `else` block.